crates/mipsy_web/src/components/data_segment.rs

```diff
@@ -1,20 +1,178 @@
+use mipsy_lib::GLOBAL_BOT;
+use mipsy_lib::KDATA_BOT;
+use mipsy_lib::KTEXT_BOT;
+use mipsy_lib::STACK_BOT;
+use mipsy_lib::STACK_TOP;
+use mipsy_lib::Safe;
+use mipsy_lib::TEXT_BOT;
+use mipsy_lib::compile::TEXT_TOP;
+use mipsy_lib::runtime::PAGE_SIZE;
 use yew::prelude::*;
 use yew::Properties;
 use crate::pages::main::state::RunningState;
 
 #[derive(Properties, Clone, PartialEq)]
-pub struct DataSegmentProps{
-    pub state: RunningState
+pub struct DataSegmentProps {
+    pub state: RunningState,
 }
 
-
+// TODO(shreys): Make this user-configurable
+fn should_display_segment(segment: Segment) -> bool {
+    match segment {
+        Segment::None =>  false,
+        Segment::Text =>  false,
+        Segment::Data =>  true,
+        Segment::Stack => true,
+        Segment::KText => false,
+        Segment::KData => false,
+    }
+}
 
 #[function_component(DataSegment)]
 pub fn data_segment(props: &DataSegmentProps) -> Html {
-    let pages = props.state.Runtime.state.pages;
+    let mut pages = props.state.mips_state.memory.iter()
```

Comment:
when a file is loaded and we are in `State::Running(...)` - pages is initially empty.

We should change the `CompileCode` response to also include the starting data